tests/integration/molecule/module_secret/converge.yml

```diff
@@ -0,0 +1,173 @@
+          - "result.object.metadata.name == 'sample-secret'"
+          - "result.object.id == 'MY_NEW_ENV_VAR'"
+          - "result.object.provider == 'env'"
+
+    - name: Make sure secret did not change
+      sensu.sensu_go.secret_info:
+        name: sample-secret
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result.objects | length == 1
+          - "result.objects.0.metadata.name == 'sample-secret'"
+          - "result.objects.0.id == 'MY_ENV_VAR'"
+          - "result.objects.0.provider == 'env'"
+
+    - name: Update existing secret
+      sensu.sensu_go.secret: *update
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result is changed
+          - "result.object.metadata.name == 'sample-secret'"
+          - "result.object.id == 'MY_NEW_ENV_VAR'"
+          - "result.object.provider == 'env'"
+
+    - name: Make sure secret is updated
+      sensu.sensu_go.secret_info:
+        name: sample-secret
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result.objects | length == 1
+          - "result.objects.0.metadata.name == 'sample-secret'"
+          - "result.objects.0.id == 'MY_NEW_ENV_VAR'"
+          - "result.objects.0.provider == 'env'"
+
+    - name: Update existing secret (idempotence)
+      sensu.sensu_go.secret: *update
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result is not changed
+
+    - name: Create additional secret
+      sensu.sensu_go.secret:
+        name: additional-secret
+        provider: hashi-vault
+        id: secret/database#pass
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result is changed
+          - "result.object.metadata.name == 'additional-secret'"
+          - "result.object.id == 'secret/database#pass'"
+          - "result.object.provider == 'hashi-vault'"
+
+    - name: Make sure we have two secrets now
+      sensu.sensu_go.secret_info:
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result.objects | length == 2
+
+    - name: Delete secret (check mode)
+      sensu.sensu_go.secret: &delete
+        name: sample-secret
+        state: absent
+      check_mode: true
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result is changed
+
+    - name: Make sure we still have two secrets
+      sensu.sensu_go.secret_info:
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result.objects | length == 2
+
+    - name: Delete secret
+      sensu.sensu_go.secret: *delete
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result is changed
+
+    - name: Now we have only one secret
+      sensu.sensu_go.secret_info:
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result.objects | length == 1
+          - "result.objects.0.metadata.name == 'additional-secret'"
+
```

Comment:
Perhaps we can add another test for  `secret_info` that verifies `result.objects` is an empty array when we try to retrieve a specific secret that does not exist.